core/frontend/src/components/vehiclesetup/overview/PowerInfo.vue

```diff
@@ -0,0 +1,158 @@
+      <span
+        v-if="batt_failsafe_low.level?.value"
+        class="d-block"
+      >
+        <b>Low Voltage Level:</b> {{ printParam(batt_failsafe_low.level) }}
+      </span>
+      <span class="d-block"><b>Critical Voltage Failsafe:</b> {{ printParam(batt_failsafe_critical.action) }}</span>
+      <span
+        v-if="batt_failsafe_critical.level?.value"
+        class="d-block"
+      >
+        <b>Low Voltage Level:</b> {{ printParam(batt_failsafe_low.level) }}
+      </span>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+import autopilot_data from '@/store/autopilot'
+import mavlink from '@/store/mavlink'
+import { printParam } from '@/types/autopilot/parameter'
+import mavlink_store_get from '@/utils/mavlink'
+
+const power_modules_list = [
+  {
+    text: 'Power Module 90A',
+    voltPin: 2,
+    currPin: 3,
+    voltMult: 10.1,
+    ampPerVolt: 17.0,
+    ampOffset: 0,
+  },
+  {
+    text: 'Power Module HV',
+    voltPin: 2,
+    currPin: 3,
+    voltMult: 12.02,
+    ampPerVolt: 39.877,
+    ampOffset: 0,
+  },
+  {
+    text: '3DR Iris',
+    voltPin: 2,
+    currPin: 3,
+    voltMult: 12.02,
+    ampPerVolt: 17.0,
+    ampOffset: 0,
+  },
+  {
+    text: 'Blue Robotics Power Sense Module',
+    voltPin: 2,
+    currPin: 3,
+    voltMult: 11.000,
+    ampPerVolt: 37.8788,
+    ampOffset: 0.330,
+  }, {
+    text: 'Navigator w/ Blue Robotics Power Sense Module',
+    voltPin: 5,
+    currPin: 4,
+    voltMult: 11.000,
+    ampPerVolt: 37.8788,
+    ampOffset: 0.330,
+  },
+]
+
+export default Vue.extend({
+  name: 'PowerInfo',
+  components: {
+  },
+  data() {
+    return {
+    }
+  },
+  computed: {
+    batt_failsafe_low() {
+      return {
+        action: autopilot_data.parameters.filter((param) => param.name === 'BATT_FS_LOW_ACT').first(),
+        level: autopilot_data.parameters.filter((param) => param.name === 'BATT_LOW_VOLT').first(),
+      }
+    },
+    batt_failsafe_critical() {
+      return {
+        action: autopilot_data.parameters.filter((param) => param.name === 'BATT_FS_CRT_ACT').first(),
+        level: autopilot_data.parameters.filter((param) => param.name === 'BATT_CRT_VOLT').first(),
+      }
+    },
+    batt_monitor() {
+      const batt_monitor = autopilot_data.parameters.filter((param) => param.name === 'BATT_MONITOR').first()
+      const batt_volt_pin = autopilot_data.parameters.filter((param) => param.name === 'BATT_VOLT_PIN').first()
+      const batt_curr_pin = autopilot_data.parameters.filter((param) => param.name === 'BATT_CURR_PIN').first()
+      const batt_volt_mult = autopilot_data.parameters.filter((param) => param.name === 'BATT_VOLT_MULT').first()
+      const batt_curr_mult = autopilot_data.parameters.filter((param) => param.name === 'BATT_AMP_PERVLT').first()
+      const batt_amp_offset = autopilot_data.parameters.filter((param) => param.name === 'BATT_AMP_OFFSET').first()
```

Comment:
you are not wrong that this is more expensive, but I like the readability, and this only runs when the BATT params change.  I CAN prefilter for BATT params first though. I'll do that.

Reply:
Could you also reduce the redundancy in the code by making a `get_parameter_by_name` function?

Reply:
yes! I'm thinking of a get_parameter_by_name thing, but on a new pr, as it will affect other things in the code.

Reply:
> I'm pretty sure it is some kind of witchcraft. I'm not 100% it is only when These parameters change, but I tested here and it ran like 3 times when I refreshed the page. (instead of 800, one for each parameter)

Haha, alrighty then :-)

Reply:
we made getters for parameters